packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -421,13 +422,18 @@ function internalSchema(
 	// Here we handle edge cases of converting the early rc/internal releases.
 	// We convert early rc releases to internal releases, and early internal releases to public releases.
 	if (prereleaseIdentifier === "rc" || prereleaseIdentifier === "dev-rc") {
-		if (semver.eq(publicVersion, "2.0.0") && semver.lt(internalVersion, "2.0.0")) {
-			if (requested === -1) {
-				return `^2.0.0-internal.8.0.0`;
+		if (semver.eq(publicVersion, "2.0.0")) {
+			// Check for most common edge case first (N-1)
```

Comment:
I think if the version math below is correct you probably don't need this special check?  Or else drop the math and only do two special casings for -1 and -2 like the other branches?  The mixed approach feels a little harder to follow